### ISSUE TITLE
Tree: the html structure has a11n problem

### ISSUE THREAD

**Issue by EugenFischer** [primefaces/primefaces#12927]
### Describe the bug

The currently rendered HTML Structure of tree component does ARIA rules.

### Reproducer

1. go to http://www.primefaces.org/showcase/ui/data/tree/contextMenu.xhtml
2. inspect page with axe accessibility browser extensions  ( [axe DevTools](https://www.deque.com/axe/browser-extensions/) )
see > "Certain ARIA roles must be contained by particular parents" violation

![grafik](https://github.com/user-attachments/assets/1f9938ed-0113-4c6a-9ae5-c345f6587a6f)

### Expected behavior

Axe Report violations for Tree fixed

Simple Example of html structure which pass the Axe Tool
```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta content="width=device-width, initial-scale=1.0" name="viewport">
  <title>Example ARIA Tree</title>
</head>
<body>
<h1>Example ARIA Tree</h1>
<section>
  <label id="treeLabel">Filesystem</label>
  <ul aria-labelledby="treeLabel" role="tree">
    <li aria-expanded="true" role="treeitem">
      <span>Folder 1</span>
      <ul role="group">
        <li role="treeitem">File 1.1</li>
        <li role="treeitem">File 1.2</li>
        <li aria-expanded="false" role="treeitem">
          <span>Folder 1.3</span>
          <ul role="group">
            <li role="treeitem">File 1.3.1</li>
            <li role="treeitem">File 1.3.2</li>
          </ul>
        </li>
      </ul>
    </li>
    <li role="treeitem">
      <span>Folder 2</span>
    </li>
  </ul>
</section>
</body>
</html>
```

### PrimeFaces edition

Community

### PrimeFaces version

13.0.10, 14.0.6

### JSF implementation

Mojarra

### JSF version

4.0.1

### Java version

17

### Browser(s)

Chrome / Edge with plugings does report the same problem

**Comment by melloware** [comment]
Thanks for this report it was definitely wrong fixed for 14.0.8

**Comment by melloware** [comment]
Test here: https://quarkus-faces-melloware-8a6a34c1.koyeb.app/ui/data/tree/basic.xhtml?jfwid=490b9
